component/topology/syncer.go

```diff
@@ -165,3 +189,29 @@ func isContextDone(ctx context.Context) bool {
 	}
 	return false
 }
+
+// ServerInfo is server static information.
+// It will not be updated when server running.
+// So please only put static information in ServerInfo struct.
+type ServerInfo struct {
+	GitHash        string `json:"git_hash"`
+	IP             string `json:"ip"`
+	Port           uint64 `json:"listening_port"`
+	StartTimestamp int64  `json:"start_timestamp"`
+}
+
+func getServerInfo() *ServerInfo {
+	info := &ServerInfo{
+		GitHash:        printer.NGMGitHash,
+		IP:             "",
+		Port:           0,
+		StartTimestamp: time.Now().Unix(),
+	}
+	cfg := config.GetGlobalConfig()
+	host, port, err := net.SplitHostPort(cfg.AdvertiseAddress)
```

Comment:
Is it possible that `AdvertiseAddress` is not set? Are there default value fallbacks?

Reply:
if `AdvertiseAddress ` is not set, it value will be `Adress`, the `Address` default value is `0.0.0.0:12020`

Reply:
Cool, there is a fallback. However this fallback looks buggy: this advertise address is just invalid.